content/articles/shell-script-to-demonstrate-the-se-of-shell-function-library/index.md

```diff
@@ -0,0 +1,188 @@
+
+To execute these tasks, some pieces of code can be used repeatedly in the program. To make this process easy and reduce errors, such codes are grouped into a `Function`, which allows them to be called and reused multiple times within the program.
+
+### How to create a function
+A function is a block of code that carries out a certain activity. A function can be Called and reused. In shell scripting, functions are analogous to other programming languages' subroutines, procedures, and functions.
+
+Simply enter the following code to create a function:
+
+```bash
+function_name () { 
+   list of commands
+}
+```
+
+Your function's name is `function_name`, and you'll call it by that name throughout your scripts. The name of the function must be preceded by parentheses and a series of instructions surrounded by braces.
+
+Below is an example to show the use of a function:
+
+```bash
+Hello () {
+   echo "Hello World"
+}
+
+Hello
+``` 
+
+Save the script above in your Linux system as `test.sh` and then run it in the terminal by writing `./test.sh`
+
+The script above will display the following output.
+
+```bash
+Hello World
+```
+
+In C/C++, a Shell Function Library is similar to a header file, whereas, in Python, it is similar to a module. The current script must be aware of the function's location. 
+
+To be able to use the functions, we must additionally add the file's location to the Environment variables, or run the script in the terminal to temporarily load the function library into the current shell. Shell Script Functions are similar to regular shell scripts in that they only define functions.
+
+### Why use the function library
+The most important reason for having a function library is for reusability. We require functions in a function library for the following reasons.
+
+1. Sometimes we require logic to address [looping logic](https://docs.oracle.com/cd/E13214_01/wli/docs102/bpguide/bpguideWhile.html). For example, in Oracle applications, whenever a concurrent request is sent, we usually click on the Refresh button to retrieve the most recent data about the concurrent program's status. Or, we may create a function and place it in the function library so it can be used in other scripts.
+
+2. We could need a collection of functions to deal with an excel file, a CSV file, an XML file, or a properties file at times. This is the situation when we wish to work with several test automation scripts. 
+
+When the functions are developed, they can be reused wherever they are needed in several test automation scripts. We may need to interact with these files for a variety of reasons, including test data iterations, spreadsheet verification, and so on.
+
+3. Some businesses even create functions for each business function of the application under test, which can then be utilized to create other test script combinations.

```

Comment:
Rephrase `businesses even create functions for each business function of the application under test`